### ISSUE TITLE
Fixes for "infinity right side" with long text in the "colspan" table cell and  unwanted '%' and '\n' when an '\anchor' directive was used inside <PRE></PRE> block

### ISSUE THREAD

**Issue by GeorgKZ** [doxygen/doxygen#10181]
1) "infinity right side"1) "infinity right side" with long text in the "colspan" table cell
in such source:

<table>
<tr> <td>col1</td>  <td>col2</td>  <td>col3</td> <td>col4</td>
<td>col5</td></tr>
<tr> <td>col1</td>  <td colspan="4">
B B B B B B B B B B B B B B B B B B B B B B B B B B B B B B B B B 
B B B B B B B B B B B B B B B B B B B B B B B B B B B B B B B B B
B B B B B B B B B B B B B B B B B B B B B B B B B B B B B B B B B
</td></tr>
</table>

"B"-line will go away to the right without line breaks.

Quick fix by inserting invisible wrapping table
@@ -1406,6 +1406,11 @@ void LatexDocVisitor::operator()(const DocHtmlCell &c)
     setCurrentColumn(currentColumn()+cs-1);
   }
 
+  if (inColSpan())
+  {
+    m_t << "{\\begin{tabularx}{\\linewidth}{*{1}{X}}";
+  }
+
   visitChildren(c);
 
   if (c.isHeading())
@@ -1417,6 +1422,12 @@ void LatexDocVisitor::operator()(const DocHtmlCell &c)
     setInRowSpan(FALSE);
     m_t << "}";
   }
+
+  if (inColSpan())
+  {
+    m_t << "\\\\\\end{tabularx}}";
+  }
+
   if (inColSpan())
   {
     setInColSpan(FALSE);

2) unwanted '%' and '\n' when an '\anchor' directive was used inside <PRE></PRE> block.

Quick fix by removing '%' and '\n' from output.
@@ -529,11 +529,11 @@ void LatexDocVisitor::operator()(const DocVerbatim &s)
 void LatexDocVisitor::operator()(const DocAnchor &anc)
 {
   if (m_hide) return;
-  m_t << "\\label{" << stripPath(anc.file()) << "_" << anc.anchor() << "}%\n";
+  m_t << "\\label{" << stripPath(anc.file()) << "_" << anc.anchor() << "}";
   if (!anc.file().isEmpty() && Config_getBool(PDF_HYPERLINKS))
   {
     m_t << "\\Hypertarget{" << stripPath(anc.file()) << "_" << anc.anchor()
-      << "}%\n";
+      << "}";
   }
 } with long text in the "colspan" table cell
in such source:

<table>
<tr> <td>col1</td>  <td>col2</td>  <td>col3</td> <td>col4</td>
<td>col5</td></tr>
<tr> <td>col1</td>  <td colspan="4">
B B B B B B B B B B B B B B B B B B B B B B B B B B B B B B B B B 
B B B B B B B B B B B B B B B B B B B B B B B B B B B B B B B B B
B B B B B B B B B B B B B B B B B B B B B B B B B B B B B B B B B
</td></tr>
</table>

"B"-line will go away to the right without line breaks.

Quick fix by inserting invisible wrapping table
@@ -1406,6 +1406,11 @@ void LatexDocVisitor::operator()(const DocHtmlCell &c)
     setCurrentColumn(currentColumn()+cs-1);
   }
 
+  if (inColSpan())
+  {
+    m_t << "{\\begin{tabularx}{\\linewidth}{*{1}{X}}";
+  }
+
   visitChildren(c);
 
   if (c.isHeading())
@@ -1417,6 +1422,12 @@ void LatexDocVisitor::operator()(const DocHtmlCell &c)
     setInRowSpan(FALSE);
     m_t << "}";
   }
+
+  if (inColSpan())
+  {
+    m_t << "\\\\\\end{tabularx}}";
+  }
+
   if (inColSpan())
   {
     setInColSpan(FALSE);

2) unwanted '%' and '\n' when an '\anchor' directive was used inside <PRE></PRE> block.

Quick fix by removing '%' and '\n' from output.
@@ -529,11 +529,11 @@ void LatexDocVisitor::operator()(const DocVerbatim &s)
 void LatexDocVisitor::operator()(const DocAnchor &anc)
 {
   if (m_hide) return;
-  m_t << "\\label{" << stripPath(anc.file()) << "_" << anc.anchor() << "}%\n";
+  m_t << "\\label{" << stripPath(anc.file()) << "_" << anc.anchor() << "}";
   if (!anc.file().isEmpty() && Config_getBool(PDF_HYPERLINKS))
   {
     m_t << "\\Hypertarget{" << stripPath(anc.file()) << "_" << anc.anchor()
-      << "}%\n";
+      << "}";
   }
 }

**Comment by albert-github** [comment]
Looks like th e"quick fix" provokes an new problem.
when having
```
Latex table test
====

HTML table
----------

<table>
<tr> <td>col1</td>  <td>col2</td>  <td>col3</td> <td>col4</td>
<td>col5</td></tr>
<tr> <td>col1</td>  <td colspan="4">
B B B B B B B B B B B B B B B B B B B B B B B B B B B B B B B B B
B B B B B B B B B B B B B B B B B B B B B B B B B B B B B B B B B
B B B B B B B B B B B B B B B B B B B B B B B B B B B B B B B B B
</td></tr>
</table>

HTML table
----------

<table>
<tr> <td>col1</td>  <td>col2</td>  <td>col3</td> <td>col4</td>
<td>col5</td></tr>
<tr> <td colspan="4">col1</td>  <td>
B B B B B B B B B B B B B B B B B B B B B B B B B B B B B B B B B
B B B B B B B B B B B B B B B B B B B B B B B B B B B B B B B B B
B B B B B B B B B B B B B B B B B B B B B B B B B B B B B B B B B
</td></tr>
</table>
```
i.e.
- first example is the example from the issue
- second example is another example where the first column has the colspan

in the **1.9.7** version this gives:
![image](https://github.com/doxygen/doxygen/assets/5223533/132c2f2a-89b6-495f-9b3d-9e686b2d2331)

running this witrh the "fix" gives:
![image](https://github.com/doxygen/doxygen/assets/5223533/4359b266-1251-44d1-8312-79bf699fc911)


So in the original example the first table is wrong and the second is OK and with the fix it is the other way around.

Example: [example.tar.gz](https://github.com/doxygen/doxygen/files/12060976/example.tar.gz)


The second problem mentioned in the issue (`<pre>`).
- I don't think this is related to the given example
- strange fix in my opinion  as the `%\n` should just signal the end of a line.

so:
- Can you please attach a, small, self contained example (source+configuration file in a, compressed, tar or zip file) that allows us to reproduce the problem? Please don't add external links as they might not be persistent.
- Please also specify the **full** doxygen version used (`doxygen -v`).

**Comment by albert-github** [comment]
> Sorry,it was the **absolutely another"** issue:

This means don't mix it with another issue but please create a separate issue for it.

**Comment by albert-github** [comment]
The changes for the table are hard to follow, it is better that you create a  PR for it.

I tried the fixes you proposed, well at least what I think you proposed) with the example as attached: [example2.tar.gz](https://github.com/doxygen/doxygen/files/12094035/example2.tar.gz)

and got:

![image](https://github.com/doxygen/doxygen/assets/5223533/40d07e6e-6677-4552-942e-ffe77e2e847a)

Doesn't look like to be correct.
